Handle exceptions in constructor

I am writing a "dirty" class for handling a test configuration

```java
    @Builder(style = BuilderStyle.STAGED_PRESERVING_ORDER)
    public FileTestConfiguration(
            Path tempDir,
            FilePreferences filePreferences,
            String sourceDir,
            String sourceFile,
            String targetDir,
            Boolean shouldStoreFilesRelativeToBibFile,
            Boolean shouldAdjustOrCopyLinkedFilesOnTransfer
    ) throws IOException {
```

The constructor also creates directories etc.

I get

```
generated\sources\annotationProcessor\java\test\org\jabref\logic\externalfiles\FileTestConfigurationBuilder.java:72: error: unreported exception IOException; must be caught or declared to be thrown
    return new FileTestConfiguration(tempDir, filePreferences, sourceDir, sourceFile, targetDir, shouldStoreFilesRelativeToBibFile, shouldAdjustOrCopyLinkedFilesOnTransfer);
           ^
```

Source:

```java
  public FileTestConfiguration build() {
    return new FileTestConfiguration(tempDir, filePreferences, sourceDir, sourceFile, targetDir, shouldStoreFilesRelativeToBibFile, shouldAdjustOrCopyLinkedFilesOnTransfer);
  }
```

Would it be possible to support exceptions at constructors, too?

--- a/src/main/java/org/jilt/internal/AbstractBuilderGenerator.java
+++ b/src/main/java/org/jilt/internal/AbstractBuilderGenerator.java
@@ -28,10 +28,12 @@ import javax.lang.model.element.TypeElement;
 import javax.lang.model.element.TypeParameterElement;
 import javax.lang.model.element.VariableElement;
 import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
 import javax.lang.model.util.Elements;
 import java.lang.annotation.ElementType;
 import java.lang.annotation.Target;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 import java.util.stream.Collectors;
 
@@ -195,6 +197,7 @@ abstract class AbstractBuilderGenerator implements BuilderGenerator {
         MethodSpec.Builder buildMethod = MethodSpec
                 .methodBuilder(this.buildMethodName())
                 .addModifiers(Modifier.PUBLIC)
+                .addExceptions(this.targetCreationMethodThrownExceptions())
                 .returns(this.targetClassTypeName());
         if (this.builderClassNeedsToBeAbstract()) {
             buildMethod.addModifiers(Modifier.ABSTRACT);
@@ -211,6 +214,17 @@ abstract class AbstractBuilderGenerator implements BuilderGenerator {
             }
         }
         return buildMethod.build();
+    }
+
+    protected final Iterable<? extends TypeName> targetCreationMethodThrownExceptions() {
+        if (this.targetCreationMethod == null) {
+            return Collections.emptyList();
+        }
+        List<TypeName> thrownExceptions = new ArrayList<>(this.targetCreationMethod.getThrownTypes().size());
+        for (TypeMirror thrownException : this.targetCreationMethod.getThrownTypes()) {
+            thrownExceptions.add(TypeName.get(thrownException));
+        }
+        return thrownExceptions;
     }
 
     protected final String accessAttributeOfTargetClass(VariableElement attribute) {

--- a/src/main/java/org/jilt/internal/AbstractTypeSafeBuilderGenerator.java
+++ b/src/main/java/org/jilt/internal/AbstractTypeSafeBuilderGenerator.java
@@ -106,6 +106,7 @@ abstract class AbstractTypeSafeBuilderGenerator extends AbstractBuilderGenerator
         interfaceBuilder.addMethod(MethodSpec
                 .methodBuilder(this.buildMethodName())
                 .addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT)
+                .addExceptions(this.targetCreationMethodThrownExceptions())
                 .returns(buildReturnType)
                 .build());
     }

--- a/src/main/java/org/jilt/internal/FunctionalBuilderGenerator.java
+++ b/src/main/java/org/jilt/internal/FunctionalBuilderGenerator.java
@@ -80,6 +80,7 @@ final class FunctionalBuilderGenerator extends AbstractTypeSafeBuilderGenerator 
                 .methodBuilder(this.builderFactoryMethodName())
                 .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
                 .addTypeVariables(this.builderClassTypeParameters())
+                .addExceptions(this.targetCreationMethodThrownExceptions())
                 .returns(this.targetClassTypeName());
 
         String builderVariableName = this.builderClassMethodParamName();
@@ -220,6 +221,7 @@ final class FunctionalBuilderGenerator extends AbstractTypeSafeBuilderGenerator 
                 .methodBuilder(toBuilderSuperMethod.name)
                 .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
                 .addTypeVariables(this.builderClassTypeParameters())
+                .addExceptions(this.targetCreationMethodThrownExceptions())
                 .returns(this.targetClassTypeName());
 
         String builderVariableName = this.builderClassMethodParamName();

--- a/src/test/java/org/jilt/test/data/exceptions/CheckedThrowsValue.java
+++ b/src/test/java/org/jilt/test/data/exceptions/CheckedThrowsValue.java
@@ -1,0 +1,25 @@
+package org.jilt.test.data.exceptions;
+
+import org.jilt.Builder;
+import org.jilt.BuilderStyle;
+
+import java.io.IOException;
+
+public final class CheckedThrowsValue {
+    @Builder(toBuilder = "toBuilder")
+    public CheckedThrowsValue() throws IOException {
+    }
+
+    @Builder(className = "CheckedThrowsValueStagedBuilder", style = BuilderStyle.STAGED, toBuilder = "toBuilder")
+    public CheckedThrowsValue(int intValue) throws IOException {
+    }
+
+    @Builder(className = "CheckedThrowsValueFunctionalBuilder", style = BuilderStyle.FUNCTIONAL, toBuilder = "toBuilder")
+    public static CheckedThrowsValue make() throws IOException {
+        return new CheckedThrowsValue();
+    }
+
+    public int getIntValue() {
+        throw new RuntimeException("CheckedThrowsValue.getIntValue() should never be called");
+    }
+}

